src/mono/mono/mini/interp/interp.c

```diff
@@ -3530,6 +3536,56 @@ static long total_executed_opcodes;
 
 #define LOCAL_VAR(offset,type) (*(type*)(locals + (offset)))
 
+/*
+ * Custom C implementations of the min/max operations for float and double.
+ * We cannot directly use the C stdlib functions because their semantics do not match
+ *  the C# methods in System.Math, but having interpreter opcodes for these operations
+ *  improves performance for FP math a lot in some cases.
+ */
+static float min_f (float lhs, float rhs) {
+	if (isnan(lhs))
```

Comment:
it would probably be cleaner to use the mono_ versions of these math methods

Reply:
if they match the .NET System.Math semantics, I'll use them. Will look into it

Reply:
are there mono_ versions of min and max? i didn't find any.

Reply:
I meant for `isnan` etc